riscv/mmu.h

```diff
@@ -38,6 +38,12 @@ struct tlb_entry_t {
   reg_t target_offset;
 };
 
+struct xlate_flags_t {
```

Comment:
For reasons I didn't have time to thoroughly investigate, passing this struct by value reduces performance substantially on my Mac M1.

If you make it a bitfield struct (i.e. the below), the perf problem goes away while preserving the semantics.

```
struct xlate_flags_t {
  const bool forced_virt : 1;
  const bool hlvx : 1;
  const bool lr : 1;
};
```

Reply:
Updated.

Reply:
Thanks!